### PR TITLE
Fix file info bug, fix get folder request

### PR DESCRIPTION
**Pull request by denysLavrienov**
This PR fixes bugs with redundant `get /folder` request, fix bug with selected file renaming, after which file info block opens.
Fixes #22 and #21

### CODE REVIEW

File: app/component/file-list-item.js [hunk 1]
```diff
@@ -87,14 +87,14 @@ export class FileListItem extends Component {
...
   }
 
   /**
    * Adds action for delete button.
    * @param {Function} handler
    */
   onDeleteButtonClick(handler) {
+    this._deleteButtonHandler = handler;
     const element = this._listItem.querySelector('.delete-button');
     element.addEventListener('click', handler);
```
Comment (Oleh-Stefanovych) [inline app/component/file-list-item.js]: Let' s define event handling in `render()`.

The same in other `on...` methods.

File: app/component/file-list-item.js [hunk 2]
```diff
@@ -87,14 +87,14 @@ export class FileListItem extends Component {
     if (this._type !== 'directory') {
       this._addDownloadButtonHandler();
     }
-    this.parentNode.appendChild(this._listItem);
   }
```
Comment (Oleh-Stefanovych) [inline app/component/file-list-item.js]: this._listItem.querySelector('.delete-button').
  .addEventListener('click', () => {
    this._deleteButtonHandler && this._deleteButtonHandler(`args`);
)

File: app/component/file-list-item.js [hunk 3]
```diff
@@ -209,5 +213,15 @@ export class FileListItem extends Component {
   set isFileRenaming(isFileRenaming) {
     this._isFileRenaming = isFileRenaming;
     this.render();
+    this._addHandlers();
```
Comment (Oleh-Stefanovych) [inline app/component/file-list-item.js]: Please remove.

File: app/service/actions.js [hunk 4]
```diff
@@ -49,7 +49,7 @@ export const actions = {
       store.commit('renameFile', fileId, newFileName);
       store.commit('setFileRenaming', false);
```
Comment (Oleh-Stefanovych) [inline app/service/actions.js]: 1. Let's dispatch `fetchFiles` action after `renameFile`.
2. Let's check is displaying (selected) file was renamed or removed in `then` block of `fetchFiles` action.

File: app/service/mutations.js [hunk 5]
```diff
@@ -69,4 +69,7 @@ export const mutations = {
   setCurrentFolder(state, folder) {
     state.currentFolder = folder;
   },
+  setSelectedFileName(state, fileName) {
```
Comment (Oleh-Stefanovych) [inline app/service/mutations.js]: You already have mutator `setSelectedFile`. Please remove.

File: app/component/file-info-block.js [hunk 6]
```diff
@@ -64,6 +64,7 @@ export class FileInfoBlock extends StatefulComponent {
       const closeButton = this._innerFileInfoBox.querySelector('.close-button');
       closeButton.addEventListener('click', () => {
         this._close();
+        this.store.state.selectedFile = undefined;
```
Comment (dkalinin) [inline app/component/file-info-block.js]: State should not be modified from component.

File: app/component/file-list-item.js [hunk 7]
```diff
@@ -71,41 +71,109 @@ export class FileListItem extends Component {
...
                           class="glyphicon glyphicon-download-alt"></span>
-                </button>`:``}
+                </button>` : ``}
                 
             </div>
         </div>`;
 
     if (this._type !== 'directory') {
       this._addDownloadButtonHandler();
+      this._addDropHandler();
```
Comment (dkalinin) [inline app/component/file-list-item.js]: Is this for folder?

File: app/service/mutations.js [hunk 8]
```diff
@@ -69,4 +60,9 @@ export const mutations = {
   setCurrentFolder(state, folder) {
     state.currentFolder = folder;
   },
+  setNewSelectedFile(state) {
+    if(state.selectedFile) {
+      state.selectedFile = state.files.find((file) => file.id === state.selectedFile.id);
```
Comment (dkalinin) [inline app/service/mutations.js]: Please move this logic to action. Use `setSelectedFile` mutation to set selectedFile field.

**Pull request merged**